cmd/protoc-gen-docs/htmlGenerator.go

```diff
@@ -652,55 +652,36 @@ func (g *htmlGenerator) generateComment(loc protomodel.LocationDescriptor, name
 			}
 		}
 
-		// remove comment blocks <!-- -->
+		// elide HTML comment blocks
 		for i := 0; i < len(lines); i++ {
 			commentStart := strings.Index(lines[i], "<!--")
 			if commentStart < 0 {
 				continue
 			}
+
 			commentEnd := strings.Index(lines[i][commentStart+3:], "-->")
 			if commentEnd >= 0 {
 				// strip out the commented portion
 				lines[i] = lines[i][:commentStart] + lines[i][commentEnd+3:]
-				if len(lines[i]) == 0 {
-					// remove lines that only contained comments
-					lines = append(lines[:i], lines[i+1:]...)
-				}
-				i--
-				// send it back through or process next line if it was deleted (don't increment i in either case)
+				i-- // run the line through the check again
 				continue
 			}
+
 			lines[i] = lines[i][:commentStart]
-			if len(lines[i]) == 0 {
-				// remove lines that only contained comments
-				lines = append(lines[:i], lines[i+1:]...)
-				i--
-			}
+
 			// find end
-			for i++; i < len(lines); {
-				commentEnd = strings.Index(lines[i][commentStart+3:], "-->")
+			for i++; i < len(lines); i++ {
```

Comment:
confusing logic - but not in the critical path so I don't think it really needs a rewrite...